src/main/java/core/basesyntax/BinaryString.java

```diff
@@ -7,6 +7,18 @@
      * а возвращает String с представлением этого числа в двоичном виде.
      */
     public String toBinaryString(int value) {
-        return null;
+        if (value == 0) {
+            return "0";
+        }
+        StringBuilder sb = new StringBuilder();
+        while (value != 0) {
+            if ((value & 1) == 1) {
```

Comment:
tell me please how it works: `value & 1`?

Reply:
It's a binary logic.  We are checking here is it odd or even. I learned it on codewars it should be working faster then %2 for searching. I know it works like this, but I can't describe exactly how it works on machinery Lvl.  Its a very nice looks for even digits, for example, **if(value&1){}** it will work for even.

Reply:
I don't like %2 this, so I made it using & I like this one, but if you say it is not nice I will correct it.